modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java

```diff
@@ -159,20 +161,31 @@ public AionTxReceipt callConstant(AionTransaction tx, IAionBlock block) {
                 aionHub.getRepository().getSnapshotTo(block.getStateRoot()).startTracking();
 
         try {
+            BlockDetails details = new BlockDetails(block, Collections.singletonList(tx));
             BulkExecutor executor =
-                    new BulkExecutor(
-                            Collections.singletonList(tx),
-                            block,
-                            repository,
-                            true,
-                            block.getNrgLimit(),
-                            LOG_VM);
+                new BulkExecutor(
+                    details,
+                    repository,
+                    true,
+                    true,
+                    block.getNrgLimit(),
+                    LOG_VM,
+                    getPostExecutionWork());
             return executor.execute().get(0).getReceipt();
         } finally {
             repository.rollback();
         }
     }
 
+    /**
+     * There is no post-execution work to do for any calls in this class.
+     */
+    private PostExecutionWork getPostExecutionWork() {
+        return (k, s, t, b) -> {
```

Comment:
Can fix this later, but I think it would make far more sense to instantiate the PostExecutionWork once at constructor (or static) time and return that instance all the time, instead of instantiating it every time this getter is called.